aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -398,5 +488,49 @@ Tensor& _histc_out_cuda(Tensor& result, const Tensor& self, int64_t bins, Scalar
   result.copy_(ret);
   return result;
 }
+
+std::tuple<Tensor,Tensor> _histogram_cuda_uniform_bins(
+    const Tensor& self,
+    int64_t nbins,
+    c10::optional<ArrayRef<double>> range,
+    const Tensor& weights /* optional */,
+    bool density) {
+  // Weights having a different shape from input is not supported yet. TO DO:
+  // Add support for weights broadcastable to input
+  bool has_weights = weights.defined();
+  if (has_weights)
+    TORCH_CHECK(
+        weights.sizes() == self.sizes(),
+        "histogram only supports input and weights of the same shape");
+  if (self.scalar_type() == ScalarType::Half) {
+    AT_ERROR("HalfTensor is not supported");
+  }
+  // Nondeterministic because of atomicAdd usage
```

Comment:
See:

https://github.com/pytorch/pytorch/blob/7d4c1a5ab07243f9c087bb957e79e4e6fcb5469d/aten/src/ATen/Context.h#L145

which also suggests including a reference to that note here and updating the list of nondeterministic operations. cc @kurtamohler.